### PR TITLE
fix(ui): stop request avatar after getting 404

### PR DESCRIPTION
**Pull request by wwayne**

### CODE REVIEW

File: ee/tabby-ui/components/user-avatar.tsx [hunk 1]
```diff
@@ -12,29 +14,43 @@ import {
...
+    data: avatarImageSrc,
+    isLoading,
+    error
+  } = useSWRImmutable(avatarUrl, (url: string) => {
+    if (!fetchAvatar) return undefined
+    return fetcher(url, {
+      responseFormatter: async response => {
+        const blob = await response.blob()
+        const buffer = Buffer.from(await blob.arrayBuffer())
+        return `data:image/png;base64,${buffer.toString('base64')}`
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/user-avatar.tsx:35]: what if image is jpg?

File: ee/tabby-ui/components/user-avatar.tsx [hunk 2]
```diff
@@ -12,29 +14,43 @@ import {
 } from '@/components/ui/avatar'
 import { Skeleton } from '@/components/ui/skeleton'
 
+const NOT_FOUND_ERROR = 'not_found'
+let fetchAvatar = true
```
Comment (liangfung) [inline ee/tabby-ui/components/user-avatar.tsx:18]: rename it to `shouldFetchAvatar` would be better

File: ee/tabby-ui/components/user-avatar.tsx (hunk 1 above)
Comment (wwayne) [reply ee/tabby-ui/components/user-avatar.tsx:35]: That is fine

File: ee/tabby-ui/components/user-avatar.tsx [hunk 3]
```diff
@@ -12,29 +14,43 @@ import {
...
+  const userId = data?.me.id
 
-  if (!data?.me?.email) return null
+  const avatarUrl = (userId && `/avatar/${data.me.id}`) || null
+  const {
+    data: avatarImageSrc,
+    isLoading,
+    error
+  } = useSWRImmutable(avatarUrl, (url: string) => {
+    if (!fetchAvatar) return undefined
```
[the remark this answers is not in the document]
Comment (wwayne) [reply ee/tabby-ui/components/user-avatar.tsx:30]: It doesn't work for `mutate(`/avatar/${data.me.id}`)`

File: ee/tabby-ui/components/user-avatar.tsx (hunk 2 above)
Comment (wwayne) [reply ee/tabby-ui/components/user-avatar.tsx]: updated

**Review by wsxiaoys — APPROVED** [review]

**Review by liangfung — APPROVED** [review]

**Pull request merged**